### ISSUE TITLE
Prism Solar DUO not compatible?

### ISSUE THREAD

**Issue by gigios** [persuader72/silla-prism-integration#2]
Hi, I have a Silla Prism Solar DUO, that is, with two cables. I tried to install the integration but the senstors I get are only those related to the first port, I don't have any sensor for the second port.
Is this incompatible with the two-output version of Prism or is there some manual setting I need to do?

Italian

Ciao, io ho un Silla Prism Solar DUO, ovvero con due cavi. Ho provato ad installare l'integrazione ma i senstori che mi ritrovo sono solo quelli relativi alla prima porta, non ho nessun sensore per la seconda porta.
È incompatibile con la versione con due uscite di Prism o c'è qualche impostazione manuale che devo effettuare?

**Comment by persuader72** [comment]
Yes, actually, I implemented only the single port functions i can try to add the multi port function and publish it in few days but I can't test it. So maybe you can provide to me some feedback if the update is working.

It will be very useful is you can provide to me the topics that your Prism DUO is publishing to your MQTT broker.

As an example I use the following command on my Raspberry to obtain the Prism topics output.

`mosquitto_sub -v -h 127.0.0.1 -t "prism/#" -u ***** -P *****`

**Comment by gigios** [comment]
Hi @persuader72, sorry for the delay in the answer.
I will try in the next days to 'grab' the topics of my Prism DUO as you have suggested.

Thanks for the answer

**Comment by gigios** [comment]
Hi @persuader72, sorry for this additional delay in the answer.
I was finally able to try to 'track' the Prims DUO topics. This is what I get (probably a lot of the data is replicated having been listening for a few minutes):

```
prism/0/info/temperature/core 26
prism/1/state 2
prism/1/amp 0
prism/1/wh 7600 
prism/1/pilot 32.0
prism/1/user_amp 10
prism/1/volt 232.5
prism/1/w 0
prism/1/wh_total 8331100
prism/1/error 0
prism/1/mode 1
prism/1/command/set_mode 1
prism/1/command/set_current_user 10.0
prism/2/state 1
prism/2/amp 0
prism/2/wh 5400
prism/2/pilot 32.0
prism/2/user_amp 20
prism/2/volt 232.0
prism/2/w 0
prism/2/wh_total 8331100
prism/2/error 0
prism/2/mode 1
prism/2/command/set_mode 1
prism/2/command/set_current_user 20.0
prism/1/volt 232.0
prism/2/volt 231.5
prism/1/volt 232.5
prism/2/volt 232.0
prism/1/volt 233.0
prism/1/volt 232.5
prism/2/volt 231.5
prism/2/volt 232.0
prism/2/volt 231.5
prism/1/volt 232.0
prism/2/volt 231.0
prism/1/volt 232.5
prism/1/session_time 519187
prism/2/volt 231.5
prism/2/session_time 5881
prism/2/volt 232.0
prism/1/volt 233.0
prism/1/volt 232.5
prism/1/volt 232.0
prism/2/volt 231.5
prism/1/volt 232.5
prism/1/volt 232.0
prism/1/volt 232.5
prism/2/wh 5400
prism/2/wh_total 8331100
prism/1/wh_total 8331100
prism/2/volt 232.0
prism/2/volt 231.5
prism/2/volt 232.0
prism/2/volt 231.5
prism/1/volt 233.0
prism/2/volt 232.0
prism/1/volt 232.5
prism/2/volt 231.5
prism/1/volt 233.0
prism/2/volt 232.5
prism/1/volt 232.5
prism/2/volt 232.0
prism/2/error 0
prism/1/volt 233.0
prism/1/error 0
prism/1/mode 1
prism/2/volt 232.5
prism/2/volt 232.0
prism/1/session_time 519246
prism/2/volt 232.5
prism/2/volt 232.0
prism/2/session_time 5881
prism/2/w 0
prism/2/amp 0
prism/2/state 1
prism/1/wh 7600
prism/1/volt 232.5
prism/2/volt 231.5
prism/1/volt 232.0
prism/1/volt 233.0
prism/2/volt 232.0
prism/1/volt 232.5
prism/1/volt 233.0
prism/1/volt 232.5
prism/1/volt 233.0
prism/1/volt 232.5
prism/1/volt 233.0
prism/2/volt 232.5
prism/2/volt 232.0
prism/1/session_time 519306
prism/2/session_time 5881
prism/2/volt 232.5
prism/1/volt 232.5
prism/2/volt 232.0
prism/1/volt 233.0
prism/2/volt 232.5
prism/1/volt 232.5
prism/2/volt 232.0
prism/1/volt 233.0
prism/2/volt 232.5
prism/1/volt 232.5
prism/2/volt 232.0
prism/1/volt 233.0
prism/2/volt 232.5
prism/1/volt 232.5
prism/2/volt 232.0
prism/1/volt 233.0
prism/2/volt 232.5
prism/1/volt 232.5
prism/2/volt 232.0
prism/1/volt 233.0
prism/1/session_time 519366
prism/1/volt 232.5
prism/2/session_time 5881
prism/1/w 0
prism/1/user_amp 10
prism/2/user_amp 20
prism/1/volt 233.0
prism/1/state 2
prism/1/amp 0
prism/1/pilot 32.0
prism/1/volt 232.5
prism/1/volt 232.0
prism/2/volt 231.5
prism/1/volt 232.5
prism/2/volt 232.0
prism/1/session_time 519426
prism/1/volt 232.0
prism/2/volt 231.5
prism/2/session_time 5881
prism/2/mode 1
prism/1/volt 232.5
prism/2/pilot 32.0
prism/2/volt 232.0
prism/2/volt 231.5
prism/1/volt 232.0
prism/1/volt 232.5
prism/1/volt 232.0
prism/1/volt 232.5
prism/1/volt 232.0
```

Thanks

**Comment by persuader72** [comment]
Hi 

I already made a patch the should solve the problem but i don't have published yet because I'm still testing it.
IF you want to test it you can try it installing last commit present in the main branch.

Anyway thanks for the data i will use it to do some cross checks with the code

Regards
Stefano

**Comment by gigios** [comment]
I will try in the next few days

**Comment by gigios** [comment]
First test, after the installation of the 0.4.0 version and updated with the 'main' branch, this is the current result.

Device Prism:
![immagine](https://github.com/user-attachments/assets/8a2cd587-1f6d-4c7a-83f6-a56e861b642b)

Port 1:
![immagine](https://github.com/user-attachments/assets/04ce0fce-2411-47f1-91bb-0a93d67f3ee4)

Port 2:
![immagine](https://github.com/user-attachments/assets/c0165dbb-68ee-4d4f-ad97-eb9d2bef2a3b)

temperature seems to me to be missing and the states of the sensors in the 'Prism' device are not clear.

Otherwise, it is already a good step forward!

**Comment by persuader72** [comment]
I added the temperature core sensor in the 0.5.1. 
About the prism device state:
1)  *Active connection* or *Connessione attiva* is a binary sensor stating that the mqtt server is reachable
2) *Power from utility* or *Potenza prelevata dalla rete* is the total power from the power grid network. This is only valid if you have the Prism power sensor connected near to your main electric meter.
3) *Total energy from grid* or *Energia totale dalla rete* is not a real prism sensor but is the integral of the power grid  sensor above, Is possible to remove it not checking the virtual sensors checkbox.

One question your  *Active connection* sensor is always in state of "disconnected"?

**Comment by gigios** [comment]
I tried the new version but I don't see any change.
![immagine](https://github.com/user-attachments/assets/1bbdb73e-5912-4e60-87bf-8294fcf768b2)

Regarding the temperature, I have a strange behavior, it seems that there are some connection problems.
![immagine](https://github.com/user-attachments/assets/f606eb8f-3c2e-45a4-97f5-468ff08dc708)
If I check the temperature with my “manual” sensor, everything seems fine without any connection problem:
![immagine](https://github.com/user-attachments/assets/02f7caa3-045a-47b1-b833-18b41a7d776d)

**Comment by persuader72** [comment]
Hi
thanks for the report, I think the problem is solved with version 0.5.4

Regards

**Comment by gigios** [comment]
For me it's not clear how it should work for the main 'device' because I have always the disconnected state for the connection and no values for the power (but I have charged my car).

![immagine](https://github.com/user-attachments/assets/28b3a5de-cfee-43fe-baf6-307fb65516a0)

Regards

**Comment by persuader72** [comment]
The first three entities depends from the topic prism/energy_data/power_grid which is oublished by the prism current sensor 
 placed on main current meter. 

But this is an optional component and this topic will be missing when the sensor component is missing like in your case.

IN 0.5.5 I changed the "Active connection sensor" that now use a topic that is always present (Or at least I hope) 
In future releases a will hide the other two topics that can't be computed because the power sensor is missing.

-----

Referring to your screenshot:
1. Active connection: Is a binary sensor trying to detect the presence of the prism using the most published topic.
2. Total energy from main: Is the Energy the your whole home consumed from the main network. So it is not directly correlated with you Car but with the overall energy consumed by the Home.
3. Power from main: Is the instant  power your Home is consuming from main network. This value is used to computed the previous sensor.

This three sensors are  computed  only if the current sensor of prism is enabled like in the following screenshot  

![image](https://github.com/user-attachments/assets/40e678eb-9b60-4a41-869f-94d5087accf5)

Maybe I must hide sensor 2 and 3 when the current sensor is not configured because in this case are pointless.

**Comment by gigios** [comment]
Perfect, thank you for your explanation and patience 😉
